sdk/communication/Shared/src/CommunicationRecordedTestSanitizer.cs

```diff
@@ -26,6 +34,11 @@ public override void SanitizeHeaders(IDictionary<string, string[]> headers)
         {
             if (headers.ContainsKey(HttpHeader.Names.Authorization))
             {
+                if (headers.ContainsKey(HttpHeader.Names.UserAgent) && headers[HttpHeader.Names.UserAgent].Any(x => x.Contains("Communication.Chat")))
```

Comment:
I think it still makes sense to separate the individual libraries constants and custom logic like this.

For example we could have this class as the base type and in each library if we need more customization we can override some of the functions and add the customization.

Reply:
Done, put custom logic in the libraries' own sanitizers and only kept the shared logic here.